### PR TITLE
oauth: Add useroauthaccesstoken API

### PR DESCRIPTION
**Pull request by stlaz**
API for https://github.com/openshift/enhancements/pull/507

### CODE REVIEW

**Comment by stlaz** [comment]
/retest
```
 '_output/tools/bin/controller-gen' schemapatch:manifests="./network/v1" paths="./network/v1" output:dir="/tmp/tmp.09FQpHszko"
Error: unable to parse option "output:dir=/tmp/tmp.09FQpHszko": [invalid digit '9' in octal literal (at <input>:1:10)] 
```
...

File: oauth/v1/types.go [hunk 1]
```diff
@@ -252,3 +252,22 @@ type RedirectReference struct {
 	// The name of the target that is being referred to. e.g. name of the Route.
 	Name string `json:"name" protobuf:"bytes,3,opt,name=name"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// UserOAuthAccessToken is a virtual resource to mirror OAuthAccessTokens to
+// the user the access token was issued for
+type UserOAuthAccessToken struct {
+	OAuthAccessToken `json:",inline" protobuf:"bytes,1,opt,name=oAuthAccessToken"`
```
Comment (sttts) [inline oauth/v1/types.go:261]: I don't like this protobuf "non-inlining". Isn't there real inlining without using different protobuf ids?

Comment (stlaz) [reply oauth/v1/types.go:261]: `type UserOAuthAccessToken OAuthAccessToken` could be a way

Comment (sttts) [reply oauth/v1/types.go:261]: yes

Comment (stlaz) [reply oauth/v1/types.go]: it seems to work fine for what we're doing in https://github.com/openshift/oauth-apiserver/pull/24

**Pull request merged**